scipy/stats/stats.py

```diff
@@ -366,6 +366,10 @@ def gmean(a, axis=0, dtype=None):
         dtype of a, unless a has an integer dtype with a precision less than
         that of the default platform integer. In that case, the default
         platform integer is used.
+    weights : array_like, optional
+        The weights array can either be 1-D (in which case its length must be
+        the size of `a` along the given `axis`) or of the same shape as `a`.
```

Comment:
if the shape of `weights` has a higher dimension than `a` what is the expected behavior?

Reply:
The expected behavior is to mimic `numpy.average`:

If `axis` is `None` (default):
```
TypeError: Axis must be specified when shapes of a and weights differ.
```

If `axis` is not `None`:
```
TypeError: 1D weights expected when shapes of a and weights differ.
```

So basically, 

> if the shape of `weights` has a higher dimension than `a`

an error is being raised.

Reply:
ok thanks for confirming.